src/modules/endpoint/Endpoint.js

```diff
@@ -37,63 +37,64 @@ class Endpoint {
-      // The correct data was already fetched
-      return oldPayload
-    }
+      const lastUrl = oldPayload.requestUrl
 
-    // Fetch if the data is not valid anymore or it hasn't been fetched yet
-    dispatch(startFetchAction(this.stateName))
+      if (lastUrl && lastUrl === formattedUrl) {
+        // The correct data was already fetched
+        return oldPayload
+      }
 
-    if (errorOverride) {
-      const payload = new Payload(false, null, errorOverride, formattedURL)
-      dispatch(finishFetchAction(this.stateName, payload))
-      return payload
-    }
-    if (responseOverride) {
-      const data = this.mapResponse(responseOverride, params)
-      const payload = new Payload(false, data, null, formattedURL)
+      // Fetch if the data is not valid anymore or it hasn't been fetched yet
+      dispatch(startFetchAction(this.stateName, formattedUrl))
+
+      if (errorOverride) {
+        const payload = new Payload(false, formattedUrl, null, errorOverride)
+        dispatch(finishFetchAction(this.stateName, payload))
+        return payload
+      }
+      if (responseOverride) {
+        const data = this.mapResponse(responseOverride, params)
+        const payload = new Payload(false, formattedUrl, data, null)
+        dispatch(finishFetchAction(this.stateName, payload))
+        return payload
+      }
+
+      const payload = await this.fetchData(formattedUrl, params)
       dispatch(finishFetchAction(this.stateName, payload))
       return payload
-    }
+    } catch (e) {
+      let error
+      if (e instanceof LoadingError || e instanceof ParamMissingError || e instanceof MappingError) {
+        error = e
+      } else {
+        error = new LoadingError({endpointName: this.stateName, message: e.message})
+      }
 
-    try {
-      const payload = await this.fetchData(formattedURL, params)
+      console.error(error)
+      const payload = new Payload(false, formattedUrl, null, error)
       dispatch(finishFetchAction(this.stateName, payload))
       return payload
-    } catch (e) {
-      console.error(`${LOADING_ERROR}: ${this.stateName}`)
-      return new Payload(false, null, LOADING_ERROR, formattedURL)
     }
   }
 
   async fetchData (formattedUrl: string, params: Params): Promise<Payload> {
     const response = await fetch(formattedUrl)
     if (!response.ok) {
-      throw new Error(LOADING_ERROR)
+      throw new LoadingError({endpointName: this.stateName, message: `${response.status}`})
     }
     try {
       const json = await response.json()
       const fetchedData = this.mapResponse(json, params)
-      return new Payload(false, fetchedData, null, formattedUrl)
+      return new Payload(false, formattedUrl, fetchedData, null)
     } catch (e) {
-      console.error(`${MAPPING_ERROR}: ${this.stateName}`)
-      console.error(e)
-      return new Payload(false, null, MAPPING_ERROR, formattedUrl)
+      throw (e instanceof MappingError) ? e : new MappingError(this.stateName, e.message)
```

Comment:
if you simplify this like suggested you could have a MappingError containing a MappingError, since a MappingError could have been thrown already, for example inside of the categories endpoint if a category does not have a parent. So this line is wrapping every error which is not already a MappingError into a MappingError, which is what we want.